contracts/actions/liquity/LiquityRedeem.sol

```diff
@@ -0,0 +1,120 @@
+    using TokenUtils for address;
+    using SafeMath for uint256;
+
+    struct Params {
+        uint256 lusdAmount;
+        address from;
+        address to;
+        address firstRedemptionHint;
+        address upperPartialRedemptionHint;
+        address lowerPartialRedemptionHint;
+        uint256 partialRedemptionHintNICR;
+        uint256 maxIterations;
+        uint256 maxFeePercentage;
+    }
+
+    /// @inheritdoc ActionBase
+    function executeAction(
+        bytes[] memory _callData,
+        bytes[] memory _subData,
+        uint8[] memory _paramMapping,
+        bytes32[] memory _returnValues
+    ) public payable virtual override returns (bytes32) {
+        Params memory params = parseInputs(_callData);
+
+        params.lusdAmount = _parseParamUint(
+            params.lusdAmount,
+            _paramMapping[0],
+            _subData,
+            _returnValues
+        );
+        params.from = _parseParamAddr(params.from, _paramMapping[1], _subData, _returnValues);
+        params.to = _parseParamAddr(params.to, _paramMapping[2], _subData, _returnValues);
+        params.maxFeePercentage = _parseParamUint(
+            params.maxFeePercentage,
+            _paramMapping[3],
+            _subData,
+            _returnValues
+        );
+
+        uint256 ethRedeemed = _liquityRedeem(params);
+        return bytes32(ethRedeemed);
+    }
+
+    /// @inheritdoc ActionBase
+    function executeActionDirect(bytes[] memory _callData) public payable virtual override {
+        Params memory params = parseInputs(_callData);
+
+        _liquityRedeem(params);
+    }
+
+    /// @inheritdoc ActionBase
+    function actionType() public pure virtual override returns (uint8) {
+        return uint8(ActionType.STANDARD_ACTION);
+    }
+
+    //////////////////////////// ACTION LOGIC ////////////////////////////
+
+    /// @notice Redeems ETH(wrapped) using LUSD with the target price of LUSD = 1$
+    function _liquityRedeem(Params memory _params) internal returns (uint256) {
+        if (_params.lusdAmount == type(uint256).max) {
+            _params.lusdAmount = LUSDTokenAddr.getBalance(_params.from);
+        }
+        LUSDTokenAddr.pullTokensIfNeeded(_params.from, _params.lusdAmount);
+
+        uint256 lusdBefore = LUSDTokenAddr.getBalance(address(this));
+        uint256 ethBefore = address(this).balance;
+
+        TroveManager.redeemCollateral(
+            _params.lusdAmount,
+            _params.firstRedemptionHint,
+            _params.upperPartialRedemptionHint,
+            _params.lowerPartialRedemptionHint,
+            _params.partialRedemptionHintNICR,
+            _params.maxIterations,
+            _params.maxFeePercentage
+        );
+
+        uint256 lusdAmountUsed = lusdBefore.sub(LUSDTokenAddr.getBalance(address(this)));   // It isn't guaranteed that the whole requested LUSD amount will be used
+        uint256 ethRedeemed = address(this).balance.sub(ethBefore);
```

Comment:
Maybe move this line after `lusdReturn` so the logic is more clear with `lusdAmountUsed` and `lusdReturn`